### ISSUE TITLE
Add trend graphs into result tables

### ISSUE THREAD

**Issue by AlvISsReimu** [penguin-statistics/frontend-v2#16]
### What problem does this feature solve?
Users will be able to see the trend of drop rate for each item of every stage.

### What does the proposed API look like?
In the drop rate table, show a small trend graph next to each drop rate. When user clicks on it, show a detailed one.

## Back-end API
Please go to swagger to see the 3rd and 4th API under Stage Controller on dev branch.

### API Usage and Parameters

1. Get segmented results for all items in the given stage
`GET /api/stages/{stageId}/trends`

- `interval` means how many milliseconds does each interval have. It also means the time gap between every two points on x-axis. Default value is 86400000 (1 day).

- `stageId` is required.

2. Get segmented results for a specific item in the given stage
`GET /api/stages/{stageId}/trends/items/{itemId}`

- `interval` Same as above.

- `stageId` Same as above.

- `itemId` is required.

### Return example

Both of these two APIs will return in the same structure.
For example:
```
{
  "interval": 864000000,
  "startTime": 1562668595938,
  "results": {
    "30012": [
      {
        "times": 3390,
        "quantity": 6337
      },
      {
        "times": 2734,
        "quantity": 4966
      },
      {
        "times": 756,
        "quantity": 1433
      },
      {}
    ],
    "30063": [
      {
        "times": 3390,
        "quantity": 1035
      },
      {
        "times": 2734,
        "quantity": 868
      },
      {
        "times": 756,
        "quantity": 221
      },
      {}
    ],
    "furni": [
      {
        "times": 3390,
        "quantity": 65
      },
      {
        "times": 2734,
        "quantity": 60
      },
      {
        "times": 756,
        "quantity": 24
      },
      {}
    ]
  }
}
```

- `startTime` is the earliest upload time of this stage. Please use this value and `interval` to generate all points on x-axis.

- If `itemId` is provided in the query, in `results` there will be only one key (the same as `itemId`). Otherwise it will contain all itemIds under this stage.

- The value of each itemId key is an array, containing all drop result objects for each section. The array index is the number of the section. Sometimes the result object will be empty `{}`, which means there is no drop sample in that section.

- `times` will never be 0. `quantity` may be 0.

- The response will have `LAST-UPDATE-TIME` in headers. Please ask @GalvinGao on its usage in caching.

**Comment by AlvISsReimu** [comment]
Improvement: We also want to show these trend graphs on "by item" result page.

**Comment by AsahiLuna** [comment]
> times will never be 0. quantity may be 0.

In fact, `times` may be 0 (when no one report).

> Improvement: We also want to show these trend graphs on "by item" result page.

No problem.

**Comment by GalvinGao** [comment]
fixed in pr #36
